Aquarius/Src/Aquarius/Core/Application.cpp

```diff
@@ -25,14 +27,26 @@ namespace Aquarius {
 
 	void Application::run()
 	{
+		auto time = std::chrono::high_resolution_clock::now();
 		while (!glfwWindowShouldClose(m_Window->get()))
 		{
-			glClearColor(0.2f, 0.3f, 0.3f, 1.0f);
-			glClear(GL_COLOR_BUFFER_BIT);
+			auto timeDelta = std::chrono::high_resolution_clock::now() - time;
+			time = std::chrono::high_resolution_clock::now();
+			for (const auto& layer : m_layerStack)
+			{
+				layer->updateAndRender(std::chrono::duration_cast<std::chrono::microseconds>(timeDelta).count() / 1000.0);
+			}
+			time = std::chrono::high_resolution_clock::now();
 			m_Window->OnUpdate();
 		}
 	}
 
+	void Application::PushLayer(uniquePtr<Layer> layer)
+	{
```

Comment:
The issue is that when I do it this way, I have to do `push_back<Layer>(std::make_unique<Layer>(std::move(layer))) which actually creates a base layer object using the implicit copy constructor, and then we lose the ability to call the overridden methods, as we actually now have a base Layer object pointed to by a Layer pointer, instead of a SandboxLayer pointed to by a Layer pointer